documentation/general/design/dash-holistic-view.md

```diff
@@ -0,0 +1,617 @@
+
+### Switch State Service (SWSS)
+
+The SWSS container comprises many daemons which operate on conceptual SONIC
+config objects across several databases. DASH affects the following daemons, as
+follows:
+
+- `orchagent`, which translates `XX_DB` objects (application and state DBs -
+  **TODO** - identify) into `ASIC_DB` objects, must be enhanced to manage new
+  DASH overlay objects, such as ACL1,2,3 rules, ENI mappings, etc. The
+  `orchagent` has to manage both representations of SONiC objects (`XX_DB` and
+  `ASIC_DB`) and translates between them bidirectionally as required.
+- `syncd`, which translates `ASIC_DB` conceptual objects into technology
+  supplier SAI library API calls, must likewise be enhanced to handle new DASH
+  SAI objects.
+
+### Switch Abstraction Interface (SAI) DASH extension
+
+The Switch Abstraction Interface (SAI) is a common API that is supported by many
+switch ASIC technology suppliers. SONiC uses SAI to program the ASIC. This
+enables SONiC to work across multiple ASIC platforms naturally. DASH uses a
+combination of traditional SAI headers and new DASH pipeline-specific extension headers.
+Technology suppliers must implement this interface for their DASH devices. This
+is the primary integration point of DASH devices and the SONiC stack. It will be
+rigorously tested for performance and conformance. See [DASH Testing
+documentation](https://github.com/Azure/DASH/tree/main/test).
+
+SAI "schema" are represented as fixed c-language header files and derived
+metadata header files. The underlay and overlay schema have different origins:
+
+- Traditional SAI headers are defined in the [OCP SAI project
+  repo](https://github.com/opencomputeproject/SAI/tree/master/inc).These are
+  hand-generated and maintained. DASH uses a subset of these to manage underlay
+  functions, e.g. device management, Layer 3 routing and so forth.
+- DASH SAI "overlay" objects are derived from a [P4 Behavioral
+  Model](https://github.com/Azure/DASH/tree/main/sirius-pipeline). A script
+  reads the P4 model and generates SAI header files.
+
+DASH uses an **enhanced syncd** to configure the dataplane using the technology
+supplier-specific SAI library.
+
+### ASIC Drivers
+
+The term "ASIC Drivers" is borrowed from traditional SONiC and SAI, where a
+datacenter switch was implemented almost entirely inside an ASIC (or multiple
+ASICs). These devices are programmed using a technology supplier Software
+Development Kit (SDK) which includes device drivers, kernel modules, etc.
+
+A contemporary DASH "SmartNIC" may consist of many complex hardware components
+including multi-core System On A Chip (SoC) ASICs, and the associated software.
+For simplicity, the software for such systems which interfaces to the SAI layer
+is collectively called the "ASIC driver." More importantly, the technology
+supplier SAI library will hide all details and present a uniform interface.
+
+
+Dedicated group was looking at PINS (which competes w/DASH). 
```

Comment:
```suggestion
Note:  a dedicated group has been investigating the use of PINS.  PINS is based upon a specific language (P4), and DPUs on the market would need to build a cross compiler. PINS places the onus onto the Cloud Providers to code and string together the PINS API and their own implementation to create the actions they want to achieve; however it may not be optimal for performance reasons in our project.
```